app/controllers/csv_controller.rb

```diff
@@ -0,0 +1,31 @@
+# app/controllers/csv_controller.rb
+class CsvController < ApplicationController
+  require 'csv'
+
+  def upload
+    if params[:csv_file].present?
+      file = params[:csv_file].path
+      parse_csv(file)
```

Comment:
This throws a fatal error and crashes the application if you upload something that isn't a `.csv` file. Replace this with exception handling to prevent this.
```
      begin
        parse_csv(file)
        flash[:notice] = "CSV file uploaded successfully."
      rescue CSV::MalformedCSVError => e
        flash[:error] = "Cannot parse csv file: #{e.message}"
      end
      ```

Reply:
Included the logic